Add ob-julia-vterm recipe

shg opened the pull request:
### Brief summary of what the package does

Provides Org-babel support for Julia code blocks using julia-vterm.

### Direct link to the package repository

https://github.com/shg/ob-julia-vterm.el

### Your association with the package

I am the author and the maintainer.

### Relevant communications with the upstream package maintainer

None needed.

### Checklist

<!-- Please confirm with `x`: -->

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [ ] I have confirmed some of these without doing them

<!-- After submitting, please fix any problems the CI reports. -->

riscy commented:
I am aware of org's `ob` == `org-babel` convention in packages, so we can disregard the package-lint nags.  Just a couple small fixes re: the metadata --

- Your summary line contains a small typo: Babel Fucntions for Julia in VTerm 
- I believe the `;;; Usage:` section in your commentary will be detected as distinct from your `;;; Commentary` (you can test this by running <kbd>M-: (lm-commentary)</kbd> in the buffer) 

And this excerpt looks like something you should place in your Commentary rather than automatically imposing it on the user, because the alias is being created outside of your implicit namespace:
```
(unless (fboundp 'org-babel-execute:julia)
  (defalias 'org-babel-execute:julia 'org-babel-execute:julia-vterm))
```

shg commented:
@riscy Thank you for the review. I have addressed the issues you pointed out. I thought the babel language name is a different concept from the emacs lisp namespace and ob-NAME and provided language name are not necessarily the same. But I understood your suggestion to avoid any confusion.

riscy commented:
All set here, thanks!

shg commented:
Thank you so much, riscy.
